c10/util/C++17.h

```diff
@@ -294,6 +294,15 @@ template<class T> inline std::string to_string(T value) {
     return detail::to_string_<T>::call(value);
 }
 
+template <class T>
+constexpr const T& min(const T& a, const T& b) {
```

Comment:
this is constexpr in C++14, right?  Is there a reason we aren't tracking C++14 / C++17 separately so we can kill the C++14 stuff once we move to that?